### PR TITLE
Added Python Solution to Cow Evolution (Bronze)

### PR DESCRIPTION
**Pull request by ryanchou-dev**
all test cases pass for this solution.

### CODE REVIEW

File: solutions/usaco-941.mdx [hunk 1]
```diff
@@ -2,9 +2,74 @@
...
+		for j in range(i + 1, len(allchars)):
+			if scan(i, j):
+				print('no')
+				return 0
+	print('yes')
+	return 0
+
+def scan(a, b):
+	""""
+	If we find a cow which has the characteristic a,
```
Comment (SansPapyrus683) [inline solutions/usaco-941.mdx:49]: maybe smth like "check if the tree is evolutionarily proper relative to characteristics a and b" or something then add the original doc comment

File: solutions/usaco-941.mdx [hunk 2]
```diff
@@ -2,9 +2,74 @@
...
+
+allchars = list(allchars)
+
+def solve():
+	# iterating through every pair of characteristics.
+	for i in range(len(allchars)):
+		for j in range(i + 1, len(allchars)):
+			if scan(i, j):
+				print('no')
+				return 0
```
Comment (SansPapyrus683) [inline solutions/usaco-941.mdx:43]: just `return` instead of `return 0`

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]
i'd ask you to take the stuff outside `solve` but that would be too much work for you

**Comment by ryanchou-dev** [comment]
> i'd ask you to take the stuff outside `solve` 

got it.

**Comment by ryanchou-dev** [comment]
Added the suggestions, thanks @SansPapyrus683!

FYI: i had to add an impossible variable since i couldn't return to stop the code.

**Comment by SansPapyrus683** [comment]
> FYI: i had to add an impossible variable since i couldn't return to stop the code.

wait wdym by this

**Comment by ryanchou-dev** [comment]
> > FYI: i had to add an impossible variable since i couldn't return to stop the code.
> 
> wait wdym by this

I had to add these parts:
image
since it's going to keep executing the lines after the for loop, so for all of the test cases with no as an answer, the output would be 
`
no
yes
`
so the impossible variable prevents the yes from printing.

**Comment by maggie-j-liu** [comment]
what about `sys.exit(0)`?

**Comment by ryanchou-dev** [comment]
> what about `sys.exit(0)`?

oh, that's a better idea.

**Comment by SansPapyrus683** [comment]
uh just saying sys.exit() also works

**Comment by ryanchou-dev** [comment]
> uh just saying sys.exit() also works

it looks cleaner 👍

File: solutions/usaco-941.mdx [hunk 3]
```diff
@@ -2,9 +2,73 @@
...
+
+for i in range(n):
+	line = input().split()
+	chars = list(line[1:])
+	cows.append(chars)
+	allchars.update(chars)
+
+allchars = list(allchars)
+
+# check if the tree is evolutionarily proper relative to characteristics a and b
```
Comment (maggie-j-liu) [inline solutions/usaco-941.mdx:37]:
```suggestion
# Check if the tree is evolutionarily proper relative to characteristics a and b
```
to be consistent

File: solutions/usaco-941.mdx [hunk 4]
```diff
@@ -2,9 +2,73 @@
...
+				hasb = 1
+		if hasa and not hasb: 
+			onlya = 1
+		if hasb and not hasa:
+			onlyb = 1
+		if hasa and hasb:
+			both = 1
+	return onlya and onlyb and both
+	
+# iterating through every pair of characteristics.
```
Comment (maggie-j-liu) [inline solutions/usaco-941.mdx:65]:
```suggestion
# Iterating through every pair of characteristics.
```

File: solutions/usaco-941.mdx [hunk 5]
```diff
@@ -2,9 +2,73 @@
...
 source: USACO Bronze 2019 Open
 title: Cow Evolution
-author: Jesse Choe
+author: Jesse Choe, Ryan Chou
 ---
 
+## Solution 
 [Official Analysis](http://www.usaco.org/current/data/sol_evolution_bronze_open19.html)
 
 [Video Analysis](https://youtu.be/4YTijr0BbrA)
```
Comment (maggie-j-liu) [inline solutions/usaco-941.mdx:11]:
```suggestion
## Video Solution
<Youtube id="4YTijr0BbrA" />
```
we have a way to embed videos (idk why this is not mentioned in the MDX module?)

**Review by maggie-j-liu — CHANGES_REQUESTED** [review]

File: solutions/usaco-941.mdx (hunk 3 above)
Comment (ryanchou-dev) [reply solutions/usaco-941.mdx]: 👍

**Review by maggie-j-liu — APPROVED** [review]

**Comment by ryanchou-dev** [comment]
i think i covered all of the suggestions, did i miss anything?

**Review by SansPapyrus683 — APPROVED** [review]

**Review by dongliu0426 — APPROVED** [review]

**Comment by ryanchou-dev** [comment]
👍

**Comment by SansPapyrus683** [comment]
can mods merge alr

**Pull request merged**